### PR TITLE
1D fixed bed adsorption/desorption example  

### PR DESCRIPTION
**Pull request by chineduobiora**
## Proposed changes:
- Temperature swing adsorption of CO2 example with the IDAES 1D fixed bed model

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by chineduobiora** [comment]
Note that for this notebook a local copy of the 1DFixedBed model is currently used. It will be removed and replaced with the model from the idaes-pse libraries when the PR (https://github.com/IDAES/idaes-pse/pull/984) is approved.

File: notebook_index.yml
Comment (aostace01) [inline notebook_index.yml:175]: Should the title be in quotation marks?  (`title: "Temperature Swing Adsorption Using 1D Fixed Bed Model"`)

**Review by aostace01 — COMMENTED** [review]
This looks good so far (still have some files to go over). Some comments related to two .svg files – these should not hold up the PR’s approval, but it would be nice to have them addressed if not too laborious:
•	plant_pfd.svg: Noticed some slight inconsistency: some unit operations and streams are capitalized (e.g., Downcomer, HP/IP/LP Turbine), while others aren't (e.g., air preheater, condensate pump).
•	boiler_scpc_PFD.svg: The legend has "FM" denoting the mass flow (kg/s), however on the flowsheet the mass flow seems to be denoted by "Fm". Also, "moisture" and "HHV" are not in the legend. Lastly, the green-colored streams are not listed in the legend (and some colors that don't show up on the flowsheet are listed).

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:17]: The original paper presenting the development of the kinetic and equilibrium model for the adsorption of CO2 onto the NETL 32D sorbent should also be cited here: 

Lee, A.; Mebane, D.; Fauth, D. J.; Miller, D. C. A Model for the Adsorption Kinetics of CO2 on Amine-Impregnated Mesoporous Sorbents in the Presence of Water. Presented at the 28th International Pittsburgh Coal Conference, Pittsburgh, PA, 2011.

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:19]: If I am not mistaken, Petsc is usually spelled as PETSc.

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:50]: Some bullet points end in a period, others don't.

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:47]: Comma missing around "i.e.".

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:96]: Typo: "an" should be removed from before "Pyomo objects".

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:130]: I think the second "petsc" should be "PETSc".

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:164]: Remove points at the end of numbered list elements (for consistency)?

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:220]: Comma missing around "i.e.".

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:276]: Comma missing around "i.e.".

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:282]: Period missing at the end of the sentence.

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:650]: Here, too, I think "Petsc" and "petsc" should be "PETSc".

File: src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb
Comment (aostace01) [inline src/Examples/Flowsheets/temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb:1790]: Another "Petsc" :)

**Review by aostace01 — COMMENTED** [review]
Some typos that stood out in `temperature_swing_adsorption/TSA_CO2_example_1DFixedBed.ipynb`:

**Comment by bpaul4** [comment]
@aostace01 thank you for your review, I will look over and address your suggestions. Regarding the SVG model tags, I believe those may be determined on the `idaes-pse` flowsheet end when the tags are generated. I will investigate if I may directly edit the files in the `examples-pse` repository.

**Comment by jghouse88** [comment]
@bpaul4 Do you know why power_generation files are changing?

**Comment by bpaul4** [comment]
@jghouse88 While fixing the TSA notebook, I noticed other index issues and addressed those as well. I added the image files to the proper `docs/` folders as part of properly indexing the power generation examples, but no example or model files are changed in the `src/` directory.

**Comment by bpaul4** [comment]
In light of the transition to a new repository, I left the SVG files alone for now. We can address those in a PR to the new repository if we would like to update those later.

**Review by aostace01 — CHANGES_REQUESTED** [review]
Brandon, I changed the title of the PR from "Temperature swing adsorption example" to "1D fixed bed adsorption/desorption example" (to better convey that this example is an adsorption step followed by a desorption step, rather than a TSA cycle example). Can you please remove references to TSA from the TSA_CO2_example_1DFixedBed.ipynb, notebook_index.yml, and notebook_index.ipynb files, and where appropriate, rephrase along the lines of "adsorption/desorption example"? Other than this, everything else looks good to me. Thanks!

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
Overall, most of my comments are asking for more detail and explanation. These are examples, so we need to explain to the user what is happening and why so that they can learn to do it themselves.

Comments on the notebook (as it is too large for me to comment on specific lines):

1. Need more introduction and a summary. What does this example show, what will be learnt, and why should the reader be interested?
1. Cycle details, remove the quotation marks around component names in the mole fraction specification.
2. Import custom libraries and functions - explain a bit more about what these are and why they need to be imported separately (and not from IDAES).
3. Function "fb_model_setup" - explain why we need to build this as a function. Also, make the heading title more explanatory, e.g. "Create a Function to Build the Flowsheet".
4. Function "fb_fix_conditions" - as above.
5. Setup solver for initialization - remove commented line on "tol", and explain why "bound_push" is required (and note the limitations of this too so users can understand).
6. Adsorption simulation - I feel there should always be some descriptive text in each markdown cell (and between all code cells, even if it is just one sentence). You need to explain what you are doing and why (users are generally reading these to learn).
7. Initialize model - we are going to want to update this example very soon to use the new initialization and scaling tools (once I get people to review and approve them). This will give an official interface to the block triangularization solve, and hopefully also remove the need for bound_push (due to variable scaling).
8. PETSc integrator - need to explain more about what is going on here and how this differs from the IPOPT solve. Also, can we cut back on the amount of output shown from PETSc?
9. Plots - need to explain the plots, and point out key features.

**Review by andrewlee94 — APPROVED** [review]

**Review by aostace01 — APPROVED** [review]
Looks good - thank you, Brandon.

**Comment by andrewlee94** [comment]
@lbianchi-lbl Could you take a look at why the notebooks are failing to convert on Window Python 3.7 please?

**Pull request merged**
